language/move-lang/src/unit_test/filter_test_members.rs

```diff
@@ -0,0 +1,181 @@
+        source_definitions,
+        lib_definitions,
+    } = prog;
+
+    let lib_definitions: Vec<_> = lib_definitions
+        .into_iter()
+        .filter_map(|def| filter_tests_from_definition(&mut context, def))
+        .collect();
+
+    let source_definitions: Vec<_> = source_definitions
+        .into_iter()
+        .filter_map(|def| filter_tests_from_definition(&mut context, def))
+        .collect();
+
+    P::Program {
+        source_definitions,
+        lib_definitions,
+    }
+}
+
+fn filter_tests_from_definition(
+    context: &mut Context,
+    def: P::Definition,
+) -> Option<P::Definition> {
+    match def {
+        P::Definition::Module(m) => filter_tests_from_module(context, m).map(P::Definition::Module),
+        P::Definition::Address(attributes, loc, addr, ms) => {
+            let ms: Vec<_> = ms
+                .into_iter()
+                .filter_map(|m| filter_tests_from_module(context, m))
+                .collect();
+            if !ms.is_empty() && !should_remove_node(context.env, &attributes) {
+                Some(P::Definition::Address(attributes, loc, addr, ms))
+            } else {
+                None
+            }
+        }
+        P::Definition::Script(P::Script {
+            attributes,
+            uses,
+            constants,
+            function,
+            specs,
+            loc,
+        }) => {
+            let script_attributes = attributes
+                .iter()
+                .chain(uses.iter().flat_map(|use_decl| &use_decl.attributes))
+                .chain(function.attributes.iter())
+                .chain(specs.iter().flat_map(|spec| &spec.value.attributes));
+
+            let errors: Vec<_> = script_attributes
+                .filter_map(|attr| {
+                    let testing_attrs = test_attributes(attr);
+                    if testing_attrs.is_empty() {
+                        None
+                    } else {
+                        Some(
+                            testing_attrs
+                                .into_iter()
+                                .map(|(loc, testing_attr)| {
+                                    let msg = format!(
+                                        "Invalid testing attribute '{}' found here",
+                                        testing_attr.name()
+                                    );
+                                    (loc, msg)
+                                })
+                                .collect(),
+                        )
+                    }
+                })
+                .collect();
+
+            if errors.is_empty() {
+                Some(P::Definition::Script(P::Script {
+                    attributes,
+                    uses,
+                    constants,
+                    function,
+                    specs,
+                    loc,
+                }))
+            } else {
+                let msg = "Testing attribute(s) found in script. This is not allowed.";
```

Comment:
This is really bizarre to read. I don't think you need this separate error since you have all of the other errors. You should just say that "Testing attributes are not allowed in scripts" for all of the sub errors here. Then you don't need this awkward standalone one